server/manager/logSyncManager/logSyncHelper.ts

```diff
@@ -0,0 +1,74 @@
+import { enableLogger, getLogs, logLevels } from "@project-sunbird/logger";
+import * as _ from "lodash";
+import * as path from "path";
+import { ErrorObj, getErrorObj, ILogAPIFormat } from "./ILogSync";
+
+const MAX_LOG_LENGTH = 30; // Number of entries
+const MAX_DAYS_TO_TRACE_TIMESTAMP = 10 * 24 * 60 * 60 * 1000; // 10 days
+
+const getAllLogs = () => {
+  enableLogger({
+    logBasePath: path.join(process.env.FILES_PATH, "logs"),
+    logLevel: process.env.LOG_LEVEL as logLevels,
+    context: { src: "desktop" },
+    adopterConfig: {
+      adopter: "winston",
+    },
+  });
+  const date = new Date();
+  const fromDay = new Date(date.getTime() - MAX_DAYS_TO_TRACE_TIMESTAMP);
+  type OrderType = "asc" | "desc";
+  const options: any = {
+    fields: ["message", "timestamp", "src", "level"],
+    from: fromDay, // 10 days back
+    until: new Date(), // today
+    start: 0,
+    order: "desc" as OrderType,
+  };
+  if (process.env.LOG_LEVEL === "error") {
+    options.rows = MAX_LOG_LENGTH;
+    options.limit = MAX_LOG_LENGTH;
+  }
+  if (getLogs) {
+    getLogs(options)
+      .then((logs) => {
+        if (_.get(logs, "dailyRotateFile.length")) {
```

Comment:
There should be else otherwise child process won't send any message if the length is 0 and handle that scenario